src/Sextant.WinUI/NavigationView.xaml.cs

```diff
@@ -0,0 +1,398 @@
+        public NavigationView(IScheduler mainScheduler, IScheduler backgroundScheduler, IViewLocator viewLocator)
+        {
+            InitializeComponent();
+
+            _logger = this.Log();
+
+            BackgroundScheduler = backgroundScheduler;
+            MainThreadScheduler = mainScheduler;
+            _viewLocator = viewLocator;
+
+            _mirroredPageStack = new Stack<IViewModel?>();
+            _mirroredModalStack = new Stack<IViewModel?>();
+
+            backButton.Visibility = Visibility.Collapsed;
+
+            PagePopped = Observable
+                .FromEvent<NavigatedEventHandler, NavigationEventArgs>(
+                    handler => (_, e) => handler(e),
+                    x => mainFrame.Navigated += x,
+                    x => mainFrame.Navigated -= x)
+                .Do(_ => IsBackButtonVisible = mainFrame.CanGoBack)
+                .Where(ep => ep.NavigationMode == NavigationMode.Back)
+                .Select(ep =>
+                {
+                    if (!(ep.Content is IViewFor view))
+                    {
+                        _logger.Debug(
+                            $"The view ({ep.Content.GetType()}) does not implement IViewFor<>.  Cannot set ViewModel from a back navigation.");
+                    }
+                    else
+                    {
+                        view.ViewModel = _mirroredPageStack.Peek();
+                    }
+
+                    // Since view stack doesn't contain instances (only types), we have to store the latest viewmodel and return it on a back nav.
+                    // ep.Content contains an instance of the new view, but it may have just been created and its ViewModel property will be null.
+                    // But we want the view that was just removed.  We need to send the old view's viewmodel to IViewStackService so that the ViewModel can be removed from the stack.
+                    return _lastPoppedViewModel;
+                })
+                .Where(x => x is not null);
+
+            BackRequested.Subscribe();
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether a modal (ContentDialog) is visible.
+        /// </summary>
+        public static bool ModalVisible
+        {
+            get
+            {
+                var popups = VisualTreeHelper.GetOpenPopups(Window.Current);
+                return popups.FirstOrDefault(x => x.Child is ContentDialog) is not null;
+            }
+        }
+
+        /// <summary>
+        /// Gets the background scheduler.
+        /// </summary>
+        public IScheduler BackgroundScheduler { get; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the default back button is visible.
+        /// </summary>
+        public bool IsBackButtonVisible
+        {
+            get => (bool)GetValue(IsBackButtonVisibleProperty);
+            set => SetValue(IsBackButtonVisibleProperty, value);
```

Comment:
Exposing the setter is worthless as it's controlled by the component itself.
One might think it's a neat way to hide the default back button, except that it gets turned on again.